### PR TITLE
DAOS-15677 cq: add copyright GHA

### PR DESCRIPTION
**Pull request by daltonbohning**
Add GHA to check for copyright update.
Move core logic from update-copyright githook into check_update_copyright.sh so the logic is shared between the githook and GHA.

Skip-build: true

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by daltonbohning** [comment]
I need to remove DEBUG and the dummy change, but I left it for now so we can see how it works.

File: src/client/dfuse/pil4dfs/hook_int.h [hunk 1]
```diff
@@ -5,6 +5,8 @@
  * SPDX-License-Identifier: BSD-2-Clause-Patent
  */
 
+
+/* dummy change */
```
Comment (daltonbohning) [inline src/client/dfuse/pil4dfs/hook_int.h:9]: Notice if you view this file in the `Files changed` tab, the annotation for the copyright is inline

File: utils/cq/check_update_copyright.sh [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+       { [ "$mode" == "githook" ] &&
+         [ "$git_vercode" -ge 2030000 ] &&
+         [ "$(git diff --cached -I Copyright "$file")" = '' ]; }; then
+        echo "DEBUG: Skipping $file"
+        continue
+    fi
+    read -r y1 y2 <<< "$(sed -nre "s/^.*$regex.*$/\4 \6/p" "$file")"
+    if [[ -z $y1 ]] ; then
+        # Print warning but don't error on non-existent copyright 
+        echo "  Copyright Information not found in: $file"
```
Comment (daltonbohning) [inline utils/cq/check_update_copyright.sh:83]: It's possible to propagate warnings to GHA but I'm not sure how many files would be flagged here

File: .github/workflows/linting.yml [hunk 3]
```diff
@@ -197,6 +197,20 @@ jobs:
       - name: Run check
         run: yamllint --format github .
 
+  copyright:
```
Comment (daltonbohning) [inline .github/workflows/linting.yml:200]: Plan to make the GHA required after we're sure it's stable

File: src/client/dfuse/pil4dfs/hook_int.h (hunk 1 above)
Comment (phender) [reply src/client/dfuse/pil4dfs/hook_int.h:9]: Looks good.

File: utils/githooks/pre-commit.d/30-Jenkinsfile.sh [hunk 4]
```diff
@@ -25,6 +25,7 @@ CURL_VERBOSE=${CURL_VERBOSE:-""}
 CURL_PROXY="${CURL_PROXY:+-x }${CURL_PROXY:-}"
 CURL_OPTS="$CURL_PROXY $CURL_VERBOSE -s"
 URL="https://$HOST/pipeline-model-converter/validate"
+# shellcheck disable=SC2086
 if ! output=$(curl $CURL_OPTS -s -X POST -F "jenkinsfile=<${1:-Jenkinsfile}" "$URL"); then
```
Comment (brianjmurrell) [inline utils/githooks/pre-commit.d/30-Jenkinsfile.sh:29]: SC2086 is usually pretty easy to resolve:

```suggestion
CURL_OPTS=($CURL_PROXY $CURL_VERBOSE -s)
URL="https://$HOST/pipeline-model-converter/validate"
if ! output=$(curl "${CURL_OPTS[@]}" -s -X POST -F "jenkinsfile=<${1:-Jenkinsfile}" "$URL"); then
```

Clearly I should have done that in the first place.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Looks good.  Please re-request review once you have cleaned it up.

File: utils/githooks/pre-commit.d/30-Jenkinsfile.sh (hunk 4 above)
Comment (daltonbohning) [reply utils/githooks/pre-commit.d/30-Jenkinsfile.sh]: Actually, no. It doesn't like that either
https://github.com/daos-stack/daos/actions/runs/12164183236/job/33925207435?pr=15552

Comment (daltonbohning) [reply utils/githooks/pre-commit.d/30-Jenkinsfile.sh]: Pushed with more quotes

**Review by phender — APPROVED** [review]

**Pull request merged**